File: src/main/java/dev/abarmin/aml/dashboard/model/DashboardModel.java
package dev.abarmin.aml.dashboard.model;

import lombok.AllArgsConstructor;
import lombok.Data;

import java.util.Collection;

@Data
@AllArgsConstructor
public class DashboardModel {
  private DashboardProfile profile;
  private Collection<BlockTypeModel> availableBlocks;
  private PageModel currentPage;
}

File: aml-frontend-private/src/model/DashboardModel.ts
import type { BlockTypeModel } from "./BlockTypeModel";
import type { DashboardProfile } from "./DashboardProfile";
import type { PageModel } from "./PageModel";

export class DashboardModel {
    profile: DashboardProfile;
    availableBlocks: BlockTypeModel[];
    currentPage: PageModel;

    constructor(profile: DashboardProfile, availableBlocks: BlockTypeModel[], currentPage: PageModel) {
        this.profile = profile;
        this.availableBlocks = availableBlocks;
        this.currentPage = currentPage;
    }

}